packages/abi-utils/lib/arbitrary.ts

```diff
@@ -0,0 +1,315 @@
+    name: EventName(),
+    inputs: fc.array(EventParameter(), { maxLength: 10 }).filter(inputs => {
+      if (inputs.filter(({ indexed }) => indexed).length > 3) {
+        // only up to 3 params can be indexed
+        return false;
+      }
+
+      // names that are not blank should be unique
+      const names = inputs.map(({ name }) => name).filter(name => name !== "");
+      return names.length === new Set(names).size;
+    }),
+    anonymous: fc.boolean()
+  });
+
+export const FunctionEntry = () =>
+  fc
+    .tuple(
+      fc.record(
+        {
+          type: fc.constant("function")
+        },
+        { withDeletedKeys: true }
+      ),
+      fc.record({
+        name: FunctionName(),
+        inputs: fc.array(Parameter(), { maxLength: 10 }).filter(inputs => {
+          // names that are not blank should be unique
+          const names = inputs
+            .map(({ name }) => name)
+            .filter(name => name !== "");
+          return names.length === new Set(names).size;
+        })
+      }),
+      fc.record(
+        {
+          outputs: fc.array(Parameter(), { maxLength: 10 }).filter(outputs => {
+            // names that are not blank should be unique
+            const names = outputs
+              .map(({ name }) => name)
+              .filter(name => name !== "");
+            return names.length === new Set(names).size;
+          })
+        },
+        { withDeletedKeys: true }
+      ),
+      fc
+        .tuple(
+          fc.oneof(
+            fc.constant("pure"),
+            fc.constant("view"),
+            fc.constant("nonpayable"),
+            fc.constant("payable")
+          ),
+          fc.boolean()
+        )
+        .map(([stateMutability, includeOptionals]) =>
+          !includeOptionals
+            ? { stateMutability }
+            : {
+                stateMutability,
+                payable: stateMutability === "payable",
+                constant:
+                  stateMutability === "view" || stateMutability === "pure"
+              }
+        )
+    )
+    .map(records => records.reduce((a, b) => ({ ...a, ...b }), {}))
+    .filter(entry => {
+      const { inputs, outputs = [] } = entry as Types.FunctionEntry;
+      // names that are not blank should be unique
+      const names = [...inputs, ...outputs]
+        .map(({ name }) => name)
+        .filter(name => name !== "");
+      return names.length === new Set(names).size;
+    });
+
+export const ReceiveEntry = () =>
+  fc.record({
+    type: fc.constant("receive"),
+    stateMutability: fc.constant("payable")
+  });
+
+export const FallbackEntry = () =>
+  fc
+    .tuple(
+      fc.record({
+        type: fc.constant("fallback")
+      }),
+      fc
+        .tuple(
+          fc.oneof(fc.constant("nonpayable"), fc.constant("payable")),
+          fc.boolean()
+        )
+        .map(([stateMutability, includeOptionals]) =>
+          !includeOptionals
+            ? { stateMutability }
```

Comment:
Same comment about stateMutability but recording it here again for good measure.